From: Marcel Holtmann <marcel@holtmann.org>
Reference: 229619
Acked-by: Karsten Keil <kkeil@suse.de> 
Subject: add init_timer() for ISDN PPP CCP reset state timer


CVE-2006-5749

Call init_timer() for ISDN PPP CCP reset state timer
The function isdn_ppp_ccp_reset_alloc_state() sets ->timer.function
and ->timer.data and later on calls add_timer() with no init_timer()
ever done. The call of init_timer() is needed, because otherwise the
call of add_timer() will result in an instant death.

Signed-off-by: Marcel Holtmann <marcel@holtmann.org>

---
commit b4407c0caec0e7531dbce691f79dea1fb983a103
tree ac06d07f2c9c3e25af5c7380d7433c110f9a3ecf
parent e25db641c0e6dd49c5db24dbe154048d4a466727
author Marcel Holtmann <marcel@holtmann.org> Tue, 19 Dec 2006 15:27:00 +0100
committer Marcel Holtmann <marcel@holtmann.org> Tue, 19 Dec 2006 15:27:00 +0100

 drivers/isdn/i4l/isdn_ppp.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/drivers/isdn/i4l/isdn_ppp.c b/drivers/isdn/i4l/isdn_ppp.c
index 1726131..4e3f127 100644
--- a/drivers/isdn/i4l/isdn_ppp.c
+++ b/drivers/isdn/i4l/isdn_ppp.c
@@ -2339,6 +2339,7 @@ static struct ippp_ccp_reset_state *isdn_ppp_ccp_reset_alloc_state(struct ippp_s
 		rs->state = CCPResetIdle;
 		rs->is = is;
 		rs->id = id;
+		init_timer(&rs->timer);
 		rs->timer.data = (unsigned long)rs;
 		rs->timer.function = isdn_ppp_ccp_timer_callback;
 		is->reset->rs[id] = rs;
